### ISSUE TITLE
Extension host terminated unexpectedly

### ISSUE THREAD

**Issue by Garvnanwani** [microsoft/vscode#103396]
Issue Type: <b>Bug</b>

I am getting this Extension host terminated unsuccessfully again and again since i have upgraded to the latest version of vs code, i have tried every possible solution to this given on the web like reinstalling, going back to previous version, disabling antivirus, disabling all extensions. But still this error persists and i am not able to use any extension

VS Code version: Code 1.47.2 (17299e413d5590b14ab0340ea477cdd86ff13daf, 2020-07-15T18:22:06.216Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-3120M CPU @ 2.50GHz (4 x 2494)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|5.88GB (1.51GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (31)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.1.0
project-manager|ale|11.1.0
vscode-tailwindcss|bra|0.4.1
npm-intellisense|chr|1.3.0
bracket-pair-colorizer|Coe|1.0.61
vscode-svgviewer|css|2.0.0
vscode-eslint|dba|2.1.8
vscode-faker|dee|1.4.1
es7-react-js-snippets|dsz|2.8.2
gitlens|eam|10.2.2
vscode-html-css|ecm|0.2.3
vscode-npm-script|eg2|0.3.12
vsc-material-theme|Equ|32.8.0
vsc-material-theme-icons|equ|1.2.0
prettier-vscode|esb|5.1.3
auto-rename-tag|for|0.1.4
beautify|Hoo|1.5.0
rest-client|hum|0.24.1
rainbow-csv|mec|1.7.0
python|ms-|2020.7.96456
debugger-for-chrome|msj|4.12.9
es6-css-minify|olb|3.3.2
material-icon-theme|PKi|4.2.0
vscode-css-peek|pra|3.2.6
quicktype|qui|12.0.46
LiveServer|rit|5.6.1
swdc-vscode|sof|2.2.38
vscodeintellicode|Vis|1.2.9
babel|wmo|1.3.0
JavaScriptSnippets|xab|1.8.0
markdown-all-in-one|yzh|3.2.0

(1 theme extensions excluded)

</details>

my developer tools console 
![image](https://user-images.githubusercontent.com/36405030/88515347-332b9a00-d009-11ea-9f69-9f1978a9d295.png)

**Comment by Garvnanwani** [comment]
I have tried

Run code --disable-extensions => doesn't work
Disable System Certificates => doesn't work
Uninstall all extensions => doesn't work
Uninstall/install clean setup => doesn't work
Use native debugging => no any dump files are found
Run as Administrator => doesn't work

**Comment by alexdima** [comment]
@Garvnanwani Can you please share the extension host logs when all extensions are disabled?
1. close all instances of vs code (make sure nothin is running even in task manager)
2. run `code --disable-extensions`
3. open a new empty window (do not open a folder or a workspace)
4. if the extension host crashes, please use `F1 > Developer: Open Logs Folder`
5. attach all `exthost*` logs in there or the entire folder with all the logs if there is something relevant in there. If you're not sure what is relevant, then attach everything.

**Comment by Garvnanwani** [comment]
Sure,
After disabling all extensions, the logs were initially empty but after some time i got this error 

![image](https://user-images.githubusercontent.com/36405030/88538323-c75b2880-d02c-11ea-8552-5fb49c05026b.png)

The extension host error usually doesnt comes on opening the file, vs code works fine till i try to save the file and then it gives something like this on console (with extensions enabled)

![image](https://user-images.githubusercontent.com/36405030/88538551-2f117380-d02d-11ea-9848-b71f5c616801.png)

**Comment by alexdima** [comment]
> The extension host error usually doesnt comes on opening the file, vs code works fine till i try to save the file and then it gives something like this on console (with extensions enabled)

Wait a second. Did you really disable extensions? Can you please do precisely the following 3 steps and answer the question: 
1. close all instances of vs code (make sure nothing is running even in task manager)
2. run `code --disable-extensions`
3. open a new empty window (do not open a folder or a workspace)

Does the extension host crash?

**Comment by Garvnanwani** [comment]
No, with the extensions disabled the extension host doens't crash and the developer console is clear too

**Comment by alexdima** [comment]
That's great! It means that one of the extensions you have installed is causing the crash. Now we know that one of the 31 installed extensions is to blame. 

To reduce the number from 31 and find the culprit, it would be very helpful if you can reproduce the extension host crash and then attach the crashing extension host's log. You can find the log via:
`F1 > Developer: Open Logs Folder` or `F1 > Developer: Show Logs... > Extension Host`.

**Comment by Garvnanwani** [comment]
Yes, the extension host crashed again

I found these logs to be suspicious

[exthost.log](https://github.com/microsoft/vscode/files/4982225/exthost.log)
[renderer1.log](https://github.com/microsoft/vscode/files/4982232/renderer1.log)

The app was running smoothly for 15-20 minutes, the git commit error was present from starting but the extension host crashed automatically after 15 minutes of using the app

**Comment by Garvnanwani** [comment]
Okay, So it happened again, i noticed something this time that as soon as i disconnect my device from wifi, the extension host crashes. It happened the same way earlier also, so maybe there something can be the issue.
This time the logs were


[exthost.log](https://github.com/microsoft/vscode/files/4982711/exthost.log)
[main.log](https://github.com/microsoft/vscode/files/4982715/main.log)
[renderer1.log](https://github.com/microsoft/vscode/files/4982716/renderer1.log)

and the console showed this again 

![image](https://user-images.githubusercontent.com/36405030/88557527-f206aa80-d047-11ea-9dfb-204e601d4def.png)

**Comment by alexdima** [comment]
@Garvnanwani Here are the active extensions (from the second log) when the crash occurred:
```
aaron-bond.better-comments
alefragnani.project-manager
CoenraadS.bracket-pair-colorizer
cssho.vscode-svgviewer
dbaeumer.vscode-eslint
eamodio.gitlens
ecmel.vscode-html-css
Equinusocio.vsc-material-theme
equinusocio.vsc-material-theme-icons
esbenp.prettier-vscode
formulahendry.auto-rename-tag
HookyQR.beautify
olback.es6-css-minify
PKief.material-icon-theme
pranaygp.vscode-css-peek
ritwickdey.LiveServer
softwaredotcom.swdc-vscode
vscode.configuration-editing
vscode.css-language-features
vscode.emmet
vscode.git
vscode.github
vscode.github-authentication
vscode.html-language-features
vscode.json-language-features
vscode.merge-conflict
vscode.microsoft-authentication
vscode.search-result
vscode.typescript-language-features
```

I see that `softwaredotcom.swdc-vscode` is running when the crash occurs and I see that the error message you have seen has a stack in that. So could you please try disabling that extension, since I think that might be the root cause of the crash?

**Comment by alexdima** [comment]
Now that I searched for "Extension host crashed" in that extension's repository, it appears this is a known issue with that extension -- https://github.com/swdotcom/swdc-vscode/issues/113

**Comment by Garvnanwani** [comment]
Sure, I have uninstalled it, and I think you are right, the extension is not crashing anymore atleast for now.
Thanks for the help Sir!

**Comment by brettmstevens7** [comment]
@Garvnanwani @alexdima Sorry about this issue. We corrected for it in `v2.2.41` and above. If you could try again with that version, it would be very helpful. Thank you and sorry again for the inconvenience.

**Comment by Garvnanwani** [comment]
@brettmstevens7 Yeah, now it's working fine, thanks for the help
